engine/annotations/ade20k.py

```diff
@@ -0,0 +1,99 @@
+            y=height
+        )
+        offscreen = gpu.types.GPUOffScreen(width, height)
+
+        with offscreen.bind():
+            fb = gpu.state.active_framebuffer_get()
+            fb.clear(color=(0.0, 0.0, 0.0, 0.0))
+            
+            gpu.state.depth_test_set('LESS_EQUAL')
+            gpu.state.depth_mask_set(True)
+            with gpu.matrix.push_pop():
+                gpu.matrix.load_matrix(matrix)
+                gpu.matrix.load_projection_matrix(projection_matrix)
+
+                def render_mesh(mesh, transform, color):
+                    mesh.transform(transform)
+                    mesh.calc_loop_triangles()
+                    vertices = np.empty((len(mesh.vertices), 3), 'f')
+                    indices = np.empty((len(mesh.loop_triangles), 3), 'i')
+
+                    mesh.vertices.foreach_get("co", np.reshape(vertices, len(mesh.vertices) * 3))
+                    mesh.loop_triangles.foreach_get("vertices", np.reshape(indices, len(mesh.loop_triangles) * 3))
+                    
+                    draw_annotation(vertices, indices, color)
+                if collection is None:
+                    for object in context.object_instances:
+                        if not hasattr(object.object, 'dream_textures_ade20k') or not object.object.dream_textures_ade20k.enabled:
+                            continue
+                        try:
+                            mesh = object.object.to_mesh()
+                            if mesh is not None:
+                                render_mesh(mesh, object.matrix_world, annotation_colors[object.object.dream_textures_ade20k.annotation])
+                                object.object.to_mesh_clear()
+                        except:
+                            continue
+                else:
+                    for object in collection.objects:
+                        if not hasattr(object, 'dream_textures_ade20k') or not object.dream_textures_ade20k.enabled:
+                            continue
+                        try:
+                            mesh = object.to_mesh(depsgraph=context)
+                            if mesh is not None:
+                                render_mesh(mesh, object.matrix_world, annotation_colors[object.dream_textures_ade20k.annotation])
+                                object.to_mesh_clear()
+                        except:
+                            continue
+            result = np.array(fb.read_color(0, 0, width, height, 4, 0, 'FLOAT').to_list())
+            result[:, :, 3] = 1
+        gpu.state.depth_test_set('NONE')
+        offscreen.free()
+        e.set()
+    bpy.app.timers.register(_execute, first_interval=0)
+    e.wait()
+    return result
+
+def draw_annotation(vertices, indices, color):
+    shader = gpu.shader.from_builtin('3D_UNIFORM_COLOR')
+    batch = batch_for_shader(
+        shader, 'TRIS',
+        {"pos": vertices},
+        indices=indices,
+    )
+    shader.uniform_float("color", color)
```

Comment:
I've run into the same color bug that was previously encountered for the openpose annotation.
[Bind the shader object. Required to be able to change uniforms of this shader.](https://docs.blender.org/api/current/gpu.types.html#gpu.types.GPUShader.bind)
Turns out that `batch.draw(shader)` would bind the shader and allow the color to be properly set after. This can be similarly changed for openpose instead of using the twice drawn background workaround.
```suggestion
    shader.bind()
    shader.uniform_float("color", color)
```